### PR TITLE
Optimize the Dockerfile a bit

### PR DESCRIPTION
**Pull request by robhudson**
This attempts to reduce Docker layers by consolidating calls. E.g. multiple `ENV` calls are done in a single layer. Move `RUN` calls together as much as possible.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,58 +1,61 @@
...
 COPY docker/bin/apt-install /usr/local/bin/
-RUN apt-install default-libmysqlclient-dev libxslt1.1 libxml2
+RUN <<EOT
+apt-install default-libmysqlclient-dev libxslt1.1 libxml2
+adduser --uid 1000 --disabled-password --gecos '' --no-create-home webdev
+chown webdev:webdev /app
+EOT
 
-ARG GIT_SHA=latest
-ENV GIT_SHA=${GIT_SHA}
```
Comment (robhudson) [inline Dockerfile:48]: This 2 got moved up to the above `ENV` call with default.

File: Dockerfile [hunk 2]
```diff
@@ -1,58 +1,61 @@
...
-RUN adduser --uid 1000 --disabled-password --gecos '' --no-create-home webdev
+# Switch to non-root user before copying files
+USER webdev
 
 COPY --from=builder /venv /venv
 COPY --from=builder /app /app
-
-# Change User
-RUN chown webdev.webdev -R .
-USER webdev
```
Comment (robhudson) [inline Dockerfile:58]: Updated to switch to `webdev` user earlier which is good for security and also means the above `COPY`s will get set to the `webdev` user.

File: Dockerfile (hunk 1 above)
Comment (pmac) [reply Dockerfile:48]: But I think we still need the `ARG` line so that we can pass in the value during build

Comment (pmac) [reply Dockerfile:48]: Yeah, it's passed in here:

https://github.com/mozmeao/basket/blob/5475580353a00dbe2ee26784aadb5fa3e96fa439/docker-compose.yml#L16

Comment (robhudson) [reply Dockerfile:48]: Tested locally and added the `ARG` back in. Thanks!

**Pull request merged**